internal/token.go

```diff
@@ -791,112 +791,74 @@ func (z *Tokenizer) readUntilCloseAngle() {
 // readString reads until a JavaScript string is closed.
 func (z *Tokenizer) readString(c byte) {
 	switch c {
+	// single quote (ends on newline)
 	case '\'':
-		z.readSingleQuoteString()
+		z.readUntilChars([]byte{'\'', '\r', '\n'})
+	// double quote (ends on newline)
 	case '"':
-		z.readDoubleQuoteString()
+		z.readUntilChars([]byte{'"', '\r', '\n'})
+	// template literal
 	case '`':
-		z.readTemplateLiteralString()
+		// Note that we DO NOT have to handle `${}` here because our expression
+		// behavior already handles `{}`. Technically incorrect, but it works.
+		z.readUntilChars([]byte{'`'})
 	}
 }
 
-func (z *Tokenizer) readSingleQuoteString() {
```

Comment:
Did a bit of cleanup in the tokenizer. We had a bunch of different utilties that all basically performed the same action:

- readSingleQuoteString()
- readDoubleQuoteString()
- readTemplateLiteralString()
- readJSSingleLineComment()
- readJSMultiLineComment()
- readRegExp()

All they did was just keep scanning until a specific character was encountered. That part has just been centralized into:

```
z.readUntilChars([]byte{'\'', '\r', '\n'})
```

This will read until any of these characters are encountered, handles escaped characters, and handles errors a bit more gracefully than the previous methods (hopefully fewer panics).